src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -807,6 +777,30 @@ - (void)asyncDispatchToMatterQueue:(void (^)(chip::Controller::DeviceCommissione
     });
 }
 
+- (void)runOnChipWorkQueue:(ChipSyncWorkQueueBlock)block error:(NSError * __autoreleasing *)error
```

Comment:
I feel like the key part that should be reflected in the name here is that the block runs _sync_.  Same for the other helper.

And again, not sure about the Chip in the name...